jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import numpy.ma as ma
+
+from traitlets import List, Unicode, Bool, Int, observe
+
+from astropy.table import QTable
+from astropy.coordinates import SkyCoord
+from astroquery.sdss import SDSS
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+
+__all__ = ['Catalogs']
+
+
+@tray_registry('imviz-catalogs', label="Imviz Catalogs")
+class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "catalogs.vue"
+    catalog_items = List([]).tag(sync=True)
+    catalog_selected = Unicode("").tag(sync=True)
+    results_available = Bool(False).tag(sync=True)
+    number_of_results = Int(0).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.catalog_items = ["SDSS"]
+
+        if self.catalog_selected not in self.catalog_items:
+            self.catalog_selected = self.catalog_items[0]
+
+    @observe("catalog_selected")
+    def vue_do_search(self, *args, **kwargs):
+        # no querying occurs while the plugin has not been opened
+        if not self.plugin_opened:
+            return
+
+        # gets the current viewer
+        viewer = self.viewer.selected_obj
+
+        # nothing happens in the case there is no image in the viewer
+        if viewer.state.reference_data is None:
+            return
```

Comment:
should this reset `results_available`/`number_of_results` just in case the viewer is emptied _after_ running a search?

Reply:
I agree! I made the changes!